src/Symfony/Component/Security/Http/HttpUtils.php

```diff
@@ -109,7 +111,19 @@ public function checkRequestPath(Request $request, string $path): bool
         if ('/' !== $path[0]) {
             // Shortcut if request has already been matched before
             if ($request->attributes->has('_route')) {
-                return $path === $request->attributes->get('_route');
+                if ($path === $request->attributes->get('_route')) {
+                    return true;
+                }
+
+                if (null === $this->urlGenerator) {
+                    return false;
+                }
+
+                try {
+                    return $request->getPathInfo() === $this->urlGenerator->generate($path);
```

Comment:
The left side gives me `/second`, but the right side gives me `/symfony-issue-53736/public/index.php/second` (in the reproducer at https://github.com/ThomasLandauer/symfony-issue-53736)

Reply:
What web server do you use and how do you start/configure it? Using the Symfony CLI I don't have the `/symfony-issue-53736/public/index.php` part on the right side.

Reply:
To get the reproducer running quickly, I used my "quick-and-dirty" server which has `localhost` configured to the parent directory of `symfony-issue-53736`. So in the browser I have http://localhost/symfony-issue-53736/public/index.php
The docs probably advise against this, but it usually works, so I'm not sure if this is an edge case or not...